machine/v1/types_powervsprovider.go

```diff
@@ -0,0 +1,154 @@
+package v1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	// PowerVSResourceTypeID enum property to identify an ID type resource reference
+	PowerVSResourceTypeID PowerVSResourceType = "ID"
+	// PowerVSResourceTypeName enum property to identify an Name type resource reference
+	PowerVSResourceTypeName PowerVSResourceType = "Name"
+	// PowerVSResourceTypeRegEx enum property to identify a tags type resource reference
+	PowerVSResourceTypeRegEx PowerVSResourceType = "RegEx"
+)
+
+// PowerVSResourceType enum attribute to identify the type of resource reference
+type PowerVSResourceType string
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// PowerVSMachineProviderConfig is the type that will be embedded in a Machine.Spec.ProviderSpec field
+// for a PowerVS virtual machine. It is used by the PowerVS machine actuator to create a single Machine.
+//
+// Compatibility level 1: Stable within a major release for a minimum of 12 months or 3 minor releases (whichever is longer).
+// +openshift:compatibility-gen:level=1
+// +k8s:openapi-gen=true
+type PowerVSMachineProviderConfig struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// userDataSecret contains a local reference to a secret that contains the
+	// UserData to apply to the instance.
+	// +optional
+	UserDataSecret *corev1.LocalObjectReference `json:"userDataSecret,omitempty"`
+
+	// credentialsSecret is a reference to the secret with IBM Cloud credentials.
+	// +optional
+	CredentialsSecret *corev1.LocalObjectReference `json:"credentialsSecret,omitempty"`
+
+	// More detail about Power VS service instance
+	// https://cloud.ibm.com/docs/power-iaas?topic=power-iaas-creating-power-virtual-server
+
+	// serviceInstance is the reference to the Power VS service on which the server instance(VM) will be created.
+	// Power VS service is a container for all Power VS instances at a specific geographic region.
+	// serviceInstance can be created via IBM Cloud catalog or CLI.
+	// serviceInstance identifier (name or id) can be obtained from IBM Cloud UI or IBM Cloud cli.
+	// +kubebuilder:validation:=Required
+	ServiceInstance PowerVSResource `json:"serviceInstance"`
+
+	// image is to identify the rhcos image uploaded to IBM COS bucket which is used to create the instance.
+	// image identifier (name or id) can be obtained from IBM Cloud UI or IBM Cloud cli.
+	// +kubebuilder:validation:=Required
+	Image PowerVSResource `json:"image"`
+
+	// network is the reference to the Network to use for this instance.
+	// network identifier (name or id) can be obtained from IBM Cloud UI or IBM Cloud cli.
+	// +kubebuilder:validation:=Required
+	Network PowerVSResource `json:"network"`
+
+	// keyPairName is the name of the KeyPair to use for SSH
```

Comment:
I'd like to see more comments about the key pair.

As I understand it, we want to say something like:
```
// The key pair will be exposed to the instance via the instance metadata service.
// On boot, the OS will copy the public keypair into the authorized keys for the core user.
```